front/app/modules/commercial/user_custom_fields/admin/containers/CustomFields/RegistrationCustomFieldEdit/index.tsx

```diff
@@ -101,4 +99,9 @@ const RegistrationCustomFieldEdit = memo(
   }
 );
 
+// based on the example here https://reactrouter.com/docs/en/v6/hooks/use-outlet-context
+export const useUserCustomFieldOutletContext = () => {
+  return useOutletContext<ContextType>();
+};
+
```

Comment:
Is this really needed? Seems to complicate the code unnecessarily. We need userCustomField in the child routes, but there we also have its id as a param, right? Then we can just do the data loading in those components.

Reply:
Yeah this is a mistake, the outletContext hook doesn't work anyway as it's a hook in a class component (oops). `/RegistrationCustomFieldEdit/RegistrationCustomFieldOptions.tsx` is a functional component so I can add withRouter and get the params there, but `RegistrationCustomFieldEdit/RegistrationCustomFieldSettings.tsx` is a class component so the hook won't work. Thankfully this seems like the only spot where this is needed, but I'm not sure of the best solution. From what I can see we don't have a `getUserCustomField` data wrapper for class components, just `getUserCustomFields` and `getUserCustomFieldsOptions`. 

I'd like to avoid rewriting these as functional components and hopefully avoid writing an entire new data fetching function here, so if you can think of the most minimal solution to pass the `customField` object we have in the parent down into these children let me know

Reply:
You can add data loading like we do in other places, with const Data = ... We have the GetUserCustomField resource component for that. Or, even better is to refactor to functional component, which is almost no work here. Then you can use the resource hook, useCustomField.

Reply:
We don't have a `GetUserCustomField ` resource (there's `fields` and `fieldOptions`) so while I'd rather avoid refactoring as part of this PR in general I think it's the best solution here. On that now..